openeo/UDF.ipynb

```diff
@@ -0,0 +1,167 @@
+       "        font.as = \"font\";\n",
+       "        font.type = \"font/woff2\";\n",
+       "        font.crossOrigin = true;\n",
+       "        font.href = \"https://use.fontawesome.com/releases/v5.13.0/webfonts/fa-solid-900.woff2\"\n",
+       "        document.head.appendChild(font);\n",
+       "    }\n",
+       "    </script>\n",
+       "    <openeo-model-builder>\n",
+       "        <script type=\"application/json\">{\"id\": \"9f2830874bae410eb1fd33b84a7e4431\", \"explicit-zoom\": true, \"height\": \"400px\", \"value\": {\"process_graph\": {\"loadcollection1\": {\"process_id\": \"load_collection\", \"arguments\": {\"bands\": [\"B04\", \"B03\", \"B02\"], \"id\": \"SENTINEL2_L2A\", \"spatial_extent\": {\"west\": 5.05, \"south\": 51.21, \"east\": 5.1, \"north\": 51.23, \"crs\": \"EPSG:4326\"}, \"temporal_extent\": [\"2022-05-01\", \"2022-05-30\"]}}, \"reducedimension1\": {\"process_id\": \"reduce_dimension\", \"arguments\": {\"data\": {\"from_node\": \"loadcollection1\"}, \"dimension\": \"t\", \"reducer\": {\"process_graph\": {\"max1\": {\"process_id\": \"max\", \"arguments\": {\"data\": {\"from_parameter\": \"data\"}}, \"result\": true}}}}, \"result\": true}}}}</script>\n",
+       "    </openeo-model-builder>\n",
+       "    "
+      ],
+      "text/plain": [
+       "<openeo.rest.datacube.DataCube at 0x7f21c0162240>"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# load collection\n",
+    "\n",
+    "cube = connection.load_collection(\n",
+    "                            \"SENTINEL2_L2A\",\n",
+    "                            bands = [\"B04\", \"B03\", \"B02\"],\n",
+    "                            temporal_extent = (\"2022-05-01\", \"2022-05-30\"),\n",
+    "                            spatial_extent = {\"west\": 5.05, \"south\": 51.21, \"east\": 5.1, \"north\": 51.23,'crs': 'EPSG:4326'}\n",
+    "\n",
+    ")\n",
+    "\n",
+    "cube = cube.reduce_dimension(dimension = \"t\", reducer = \"max\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here the UDF code shown in the following cell does the actual value rescaling. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Build a UDF object from an inline string with Python source code.\n",
+    "udf = openeo.UDF(\"\"\"\n",
+    "from openeo.udf import XarrayDataCube\n",
+    "\n",
+    "def apply_datacube(cube: XarrayDataCube, context: dict) -> XarrayDataCube:\n",
+    "    array = cube.get_array()\n",
+    "    array.values = 0.0001 * array.values\n",
+    "    return cube\n",
+    "\"\"\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "User can also load their UDF from a seperate file using `openeo.UDF.from_file('my_udf.py')` and apply it along a dimension. "
```

Comment:
I would remove `and apply it along a dimension` which is confusing because in the example (which uses `apply`) you don't apply it *along a dimension*.

Reply:
updated